client/jetpack-app/plans/main.jsx

```diff
@@ -0,0 +1,109 @@
+import { getPlan, PLAN_FREE } from '@automattic/calypso-products';
+import { translate } from 'i18n-calypso';
+import { useSelector } from 'react-redux';
+import QueryPlans from 'calypso/components/data/query-plans';
+import FormattedHeader from 'calypso/components/formatted-header';
+import { LoadingEllipsis } from 'calypso/components/loading-ellipsis';
+import Main from 'calypso/components/main';
+import { getPlanCartItem } from 'calypso/lib/cart-values/cart-items';
+import PlansFeaturesMain from 'calypso/my-sites/plans-features-main';
+import { getPlanSlug } from 'calypso/state/plans/selectors';
+
+import './style.scss';
+
+const Header = ( { paidDomainName } ) => (
+	<div className="plans__header">
+		<FormattedHeader
+			brandFont
+			headerText={ translate( 'Choose the perfect plan' ) }
+			align="center"
+		/>
+		{ paidDomainName ? (
+			<>
+				<p>
+					{ translate(
+						'With your annual plan, you’ll get %(domainName)s {{strong}}free for the first year{{/strong}}.',
+						{
+							args: {
+								domainName: paidDomainName,
+							},
+							components: { strong: <strong /> },
+						}
+					) }
+				</p>
+				<p>
+					{ translate(
+						'You’ll also unlock advanced features that make it easy to build and grow your site.'
+					) }
+				</p>
+			</>
+		) : (
+			<p>{ translate( 'See and compare the features available on each WordPress.com plan.' ) }</p>
+		) }
+	</div>
+);
+
+const Plans = ( { paidDomainName, onUpgradeClick, handleRedirect } ) => (
+	<PlansFeaturesMain
+		paidDomainName={ paidDomainName }
+		intent="plans-jetpack-app-site-creation"
+		isInSignup={ true }
+		intervalType="yearly"
+		onUpgradeClick={ onUpgradeClick }
+		plansWithScroll={ false }
+		removePaidDomain={ handleRedirect }
+		hidePlanTypeSelector={ true }
+		hidePlansFeatureComparison={ true }
+	/>
+);
+
+const Loading = () => (
+	<div className="plans__loading">
+		<LoadingEllipsis active />
+	</div>
+);
+
+const JetpackAppPlans = ( { paidDomainName, redirectTo } ) => {
+	const planSlug = useSelector( ( state ) =>
+		getPlanSlug( state, getPlan( PLAN_FREE )?.getProductId() || 0 )
+	);
+	const plansLoaded = Boolean( planSlug );
```

Comment:
Thanks for letting me know about this!

There's a slight issue. I have a `<QueryPlans />` component that starts loading the Plans. If I use

```
const isPlansListLoading = useSelector( ( state: AppState ) => isRequestingPlans( state ) );
```

within `JetpackAppPlans` component, then there's a constant switching between `<QueryPlans />` triggering loading state, and then `isPlansListLoading` turning to false making the component to re-render triggering `<QueryPlans />` again.

I'm not sure if it's worth extracting `<QueryPlans />` to a parent component to avoid this? 🤔  

For now, I left the solution as it is, especially given `plansLoaded` is used in other Plans-related components.